### ISSUE TITLE
discover and validate storage options

### ISSUE THREAD

**Issue by stoneshi-yunify** [kubesphere/kubesphere#4741]
user may want to know the storage capabilities in his/her cluster, such as:
- storage classes
- csi or non-csi drivers
- if snapshot is supported
- io performance numbers
- etc.

A such useful tool needs to be developed to do this.

ref: https://github.com/kastenhq/kubestr

/kind feature-request
/area storage
/milestone backlog

**Comment by f10atin9** [comment]
I understand that these capabilities are displayed in the storageClass details page in the console, right?

**Comment by stoneshi-yunify** [comment]
> I understand that these capabilities are displayed in the storageClass details page in the console, right?

no. this issue is not related to ks console. The tool this issue mentioned aims to discover the storage options before installing ks or any other apps.

**Comment by f10atin9** [comment]
Okay， I understand the meaning.
